src/fundus/parser/utility.py

```diff
@@ -121,14 +121,23 @@ def extract_nodes(selector: XPath, node_type: Type[Node]) -> List[Node]:
 _meta_node_selector = CSSSelector("meta[name], meta[property]")
 
 
-def get_meta_content(tree: lxml.html.HtmlElement) -> Dict[str, str]:
+def get_meta_content(
+    tree: lxml.html.HtmlElement, additional_selectors: Optional[Dict[str, Union[XPath, CSSSelector]]] = None
+) -> Dict[str, str]:
     meta_nodes = _meta_node_selector(tree)
     meta: Dict[str, str] = {}
     for node in meta_nodes:
         key = node.attrib.get("name") or node.attrib.get("property")
         value = node.attrib.get("content")
         if key and value:
             meta[key] = value
+
+    if additional_selectors:
+        for key, selector in additional_selectors.items():
+            for node in selector(tree):
+                if content := node.attrib.get("content"):
+                    meta[node.attrib[key]] = content
+
```

Comment:
As it seems to me the user needs to pass the same value for the key as the meta attribute they are filtering by. e.g. calling `get_meta_content(doc, dict({"test": lxml.cssselect.CSSSelector("meta[itemprop]")}))` raises a KeyError. I was wondering whether it is really necessary to pass a dict, which causes an exception if used incorrectly, if the value of the key is already predetermined.

Reply:
Yeah, you're right. Using a dict with the current implementation doesn't make much sense. I'm gonna think of something else here after reading a bit through the documentation.